ara/exec/parser/manifest_parser.cpp

```diff
@@ -1,23 +1,228 @@
+bool ExecutionManifest::Process::StartupConfig::StartupOption::operator==(
+    const StartupOption& other) const noexcept
 {
-    // TODO implement this method
+    return (kind == other.kind) && (name == other.name) && (arg == other.arg);
+}
+
+bool ExecutionManifest::Process::StartupConfig::StartupOption::operator!=(
+    const StartupOption& other) const noexcept
+{
+    return !(*this == other);
+}
+
+bool ExecutionManifest::Process::StartupConfig::MachineInstanceRef::operator==(
+    const MachineInstanceRef& other) const noexcept
+{
+    return (function_group == other.function_group) && (mode == other.mode);
+}
+
+bool ExecutionManifest::Process::StartupConfig::MachineInstanceRef::operator!=(
+    const MachineInstanceRef& other) const noexcept
+{
+    return !(*this == other);
+}
+
+namespace {
+namespace EMJsonKeys {
+const std::string kApplicationManifest = "Application_manifest";
+const std::string kApplicationManifestId = "Application_manifest_id";
+const std::string kProcess = "Process";
+const std::string kProcessName = "Process_name";
+const std::string kModeDependentStartupConfigs = "Mode_dependent_startup_configs";
+const std::string kStartupOptions = "Startup_options";
+const std::string kStartupOptionsOptionKind = "Option_kind";
+const std::string kStartupOptionsOptionName = "Option_name";
+const std::string kStartupOptionsOptionArg = "Option_arg";
+const std::string kModeInMachineInstanceRefs = "Mode_in_machine_instance_refs";
+const std::string kFunctionGroup = "Function_group";
+const std::string kMode = "Mode";
+
+const std::vector<std::string> kAsVector{kApplicationManifest,
+                                         kApplicationManifestId,
+                                         kProcess,
+                                         kProcessName,
+                                         kModeDependentStartupConfigs,
+                                         kStartupOptions,
+                                         kStartupOptionsOptionKind,
+                                         kStartupOptionsOptionName,
+                                         kStartupOptionsOptionArg,
+                                         kModeInMachineInstanceRefs,
+                                         kFunctionGroup,
+                                         kMode};
+
+}  // namespace EMJsonKeys
+
+namespace MMJsonKeys {
+const std::string kMachineManifest = "Machine_manifest";
+// TODO add rest of the keys
+
+const std::vector<std::string> kAsVector{kMachineManifest /* TODO add rest of the keys */};
+}  // namespace MMJsonKeys
+
+}  // anonymous namespace
+
+template <typename T>
+bool try_to_read_value(const json& json_obj, const std::string& key, T& output_value) noexcept
+{
+    T tmp{};
+    try {
+        tmp = json_obj.at(key);
+    } catch (json::exception& e) {
+        std::cerr << "ara::exec::parser::try_to_read_value -> " << e.what()
+                  << '\n';  // development purposes
```

Comment:
Zbedny komentarz, wylapanie bledu mozna rozbic na out_of_range i pozostale jakos exception& e z odpowiednimi komunikatami

Reply:
Tak naprawde, nie interesuje nas rodzaj tego wyjątku. Niezależnie czy jest to `json::out_of_range`, czy `json::type_error`, jedyne co chcemy zrobic, to zwrócić `false` z tej funkcji. Komunikat błędu jest potrzebny tylko na czas pracy nad parserem- na koniec zostanie usunięty, bo prawdziwa walidacja manifestu, odbywać się będzie w innej funkcji

Reply:
Guidelines for C++14 for AUTOSAR się z Tobą nie zgodzi :(

Reply:
No ok, ale po co mam pisać coś takiego:
```
try {
...
} catch (json::out_of_range&) {
    return false;
} catch (json::type_error&) {
    return false;
}
```

Reply:
Zmienię na coś takiego, żeby wszystko było zgodnie ze standardem, ale wg mnie to jest bez sensu, bo `json::out_of_range` i `json::type_error` są jedynymi wyjątkami jakie `at` moze rzucić i obsługujemy je w ten sam sposób- zwracając `false` z funkcji, także jak dla mnie to niepotrzebne powtórzenie

Reply:
Hmm bardziej chodzilo mi o to ze w jednym z tych przypadkow bedzie throw a w drugim return false

Reply:
nie no, ideą tej funkcji jest to, żeby przy każdym czytaniu nie trzeba bylo robić `try/catch` (zobacz ile jest wywołań tej funkcji w `ManifestParser::parse_execution_manifest(...)`), tylko schludnie owrapować te bloki w funkcję, a rzucanie z tej metody wyjątkiem minie się z celem